libusb/descriptor.c

```diff
@@ -1168,9 +1168,7 @@ int API_EXPORTED libusb_get_string_descriptor_ascii(libusb_device_handle *dev_ha
 	r = libusb_get_string_descriptor(dev_handle, 0, 0, str.buf, 4);
 	if (r < 0)
 		return r;
-	else if (r != 4 || str.desc.bLength < 4)
-		return LIBUSB_ERROR_IO;
-	else if (str.desc.bDescriptorType != LIBUSB_DT_STRING)
+	else if (r != 4 || str.desc.bLength < 4 || str.desc.bDescriptorType != LIBUSB_DT_STRING)
```

Comment:
what was the warning here?
I personally find the "before" more descriptive/readable

Reply:
https://clang.llvm.org/extra/clang-tidy/checks/bugprone/branch-clone.html